layouts/fastly/helix.vcl

```diff
@@ -1270,6 +1270,10 @@ sub hlx_fetch_error {
       error 965 "Too many requests";
     } elsif (beresp.status == 400) {
       error 966 "Bad request";
+    } elsif (beresp.status == 401) {
+      // we handle a 401 from the backend just like a 403, as there won't be any
```

Comment:
I am not sure about the 401 -> 403.

Reply:
https://datatracker.ietf.org/doc/html/rfc7235#section-3.1

>   The 401 (Unauthorized) status code indicates that the request has not
>   been applied because it lacks valid authentication credentials for
>   the target resource.  The server generating a 401 response **MUST** send
>   a `WWW-Authenticate` header field (Section 4.1) containing at least one
>   challenge applicable to the target resource.

We do not have a challenge that would work in this case. It might be different for private repos and a missing `x-github-token` header, but for the scenario that triggered the bug (Sharepoint folder not shared with Helix) there is nothing the client can do.